README.md

```diff
@@ -1,123 +1,343 @@
 ```
+
+#### Parameters:
+
+* --awsRegion: The region where the AWS resources reside.
+
+* --awsCredentialsProvider: The AWS credentials provider.
+
+
+## How to adapt this pipeline for your use cases
+
+1. [Create your own inspection templates & run inspection on sample data](https://cloud.google.com/dlp/docs/creating-templates-inspect)
+2. [Create de-identification templates and run de-identification on sample data](https://cloud.google.com/dlp/docs/creating-templates-deid)
+3. In case of cross-regional data source or sink, dataflow service account should be granted necessary permissions.
+
+
+## Troubleshooting
+
+Following are the known issues with Cloud DLP, along with ways you can avoid or recover from them.
+
+1. Duplicate Rows: When writing data to a BigQuery table, Cloud DLP might write duplicate rows.
+
+    > The project uses Streaming Inserts API of BigQuery which by default enables best-effort deduplication mechanism but it should not be relied upon as a mechanism to guarantee the absence of duplicates in your data.
+    For solution, checkout [high number of duplicates in Dataflow pipeline streaming inserts to BigQuery](https://cloud.google.com/knowledge/kb/high-number-of-duplicates-in-dataflow-pipeline-streaming-inserts-to-bigquery-000004276?authuser=0).
+
+2. Error in DLP API : "Too many findings in request"
+
+   > DLP has a max findings per request [limit](https://cloud.google.com/dlp/limits#content-redaction-limits) of 3000.
+   Run pipeline again with smaller batch size. 
+
+3. DLP API request quota exhausted
+
+   > This can happen if the dataflow pipeline is being run with a small batch size. Rerun the pipeline with a larger value of batch size.
+   If batch size cannot be increased or issue persists despite saturating the batch size:
+   > 
+   >  * Increase the value of parameter dlpApiRetryCount
+   >
+   >  * The dlp-dataflow-deidentification repo offers a parameter numShardsPerDLPRequestBatching. Reducing this below the default (100) will reduce the number of parallel requests sent to DLP.
+   >
+   >  * Review if there are other pipelines/clients generating DLP API requests.  Submit a request to increase the quota.
+   >
+   >  * Submit a request to increase the quota.
+
+
 ## Dataflow DAG
 
 For Deid and Inspect:
 
-![v2_dag_](diagrams/dlp_dag_new.png)	   	
+![v2_dag_](diagrams/dlp_dag_new.png)
 
 
 
 For Reid:
 
-![v2_dag_](diagrams/dlp_reid_dag.png)	   	
+![v2_dag_](diagrams/dlp_reid_dag.png)
 
 
 
-## Trigger Pipeline Using Public Image
+## Advanced topics
+
 You can use the gcloud command to trigger the pipeline using Dataflow flex template. Below is an example for de-identification transform from a S3 bucket.
```

Comment:
Re: Below is an example for de-identification transform from a S3 bucket.

How about: The following example runs a de-identification transformation on data from an S3 bucket.